File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 6/25/2026, 9:18:11 AM
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
  
        result =0 
        d =     {'I':1,
                'V':5,
                'X':10,
                'L':50,
                'C':100,
                'D':500,
                'M':1000
                }
        s=s.replace("IV",'IIII').replace('IX','VIIII')
        s=s.replace("XC",'LXXXX').replace('CD','CCCC')
        s=s.replace("CM",'DCCCC').replace('XL','XXXX')
        for i in s:
            result+= d[i]
        return result

            

File: 13-RomanToInteger/13-RomanToInteger.cpp
// Last updated: 6/25/2026, 9:18:16 AM
class Solution {
public:
    int romanToInt(string s) {
        int num = 0;
        vector<char> roman{'I', 'V', 'X', 'L', 'C', 'D', 'M'};
        for(int i=0; i<s.size(); i++){
            if(s[i] == 'M') num += 1000;
            else if(s[i] == 'D') num += 500;
            else if(s[i] == 'C'){
                if(i == s.size()-1) num += 100;
                else if(s[i+1] =='M') {num += 900; i++;}
                else if(s[i+1] =='D') {num += 400; i++;}
                else num += 100;
            }
            else if(s[i] == 'L') num += 50;
            else if(s[i] == 'X'){
                if(i == s.size()-1) num += 10;
                else if(s[i+1] =='C') {num += 90; i++;}
                else if(s[i+1] =='L') {num += 40; i++;}
                else num += 10;
            }
            else if(s[i] == 'V') num += 5;
            else if(s[i] == 'I'){
                if(i == s.size()-1) num += 1;
                else if(s[i+1] =='X') {num += 9; i++;}
                else if(s[i+1] =='V') {num += 4; i++;}
                else num += 1;
            }
        }
        return num;
    }
};